tests/components/zwave_js/test_sensor.py

```diff
@@ -261,6 +261,41 @@ async def test_config_parameter_sensor(
     await hass.async_block_till_done()
 
 
+async def test_controller_status_sensor(
+    hass: HomeAssistant, client, integration
+) -> None:
+    """Test controller status sensor is created and gets updated on controller state changes."""
+    entity_id = "sensor.z_stick_gen5_usb_controller_status"
+    ent_reg = er.async_get(hass)
+    entity_entry = ent_reg.async_get(entity_id)
+
+    assert not entity_entry.disabled
+    assert entity_entry.entity_category is EntityCategory.DIAGNOSTIC
+    assert hass.states.get(entity_id).state == "ready"
+    assert hass.states.get(entity_id).attributes[ATTR_ICON] == "mdi:check"
```

Comment:
Store the state in a local variable first, and assert it, so we don't need to look it up more than once per round of events.